### PR TITLE
[MDS-5337] Cache fontawesome using Artifactory

### PR DESCRIPTION
**Pull request by simensma-fresh**
## Objective 

[MDS-5337](https://bcmines.atlassian.net/browse/MDS-5337)
- Switched to downloading fontawesome packages from the [Artifactory BCGov service](https://developer.gov.bc.ca/Artifact-Repositories-\(Artifactory\)) in order to get around the Fontawesome npm package bandwith limits
- Added documentation on the setup, and how to upgrade Fontawesome to newer versions when required

This includes some type-related changes. Why? When re-generating `yarn.lock` in order to change the npm repository of Fontawesome, about 100 Typescript related react errors popped up. Why? Apparently some peer-dependencies requires `@types/react` of version `*` which resolves to the latest published react types, which caused things to throw errors. Solution: Force resolution of `@types/react`  to match React version we're using

### CODE REVIEW

File: services/core-web/common/components/SteppedForm.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import { indexOf } from "lodash";
 const { Item } = Menu;
 
 const propTypes = {
-  children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  children: PropTypes.arrayOf(PropTypes.any).isRequired,
```
Comment (simensma-fresh) [inline services/core-web/common/components/SteppedForm.js:13]: For some reason I could not figure out, `PropTypes.node` would not recognize the `children` property type correctly, hence this unfortunate change

**Review by henryoforeh-dev — APPROVED** [review]

File: docs/feature_flags.md [hunk 2]
```diff
@@ -88,9 +84,9 @@ const mapDispatchToProps = (dispatch) =>
     dispatch
   );
 
-export default compose(
-  connect(mapStateToProps, mapDispatchToProps),
-)(withRouter(FeatureFlagGuard(Feature.TSF_V2)(DamsPage)));
+export default compose(connect(mapStateToProps, mapDispatchToProps))(
```
Comment (matbusby-fw) [inline docs/feature_flags.md:87]: Good catch on a needed documentation update!

File: services/core-web/common/components/Step.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { FC } from "react";
+
+interface StepProps {
+  disabled?: boolean;
+  key: string;
+}
+
+const Step: FC<StepProps> = (props) => {
```
Comment (matbusby-fw) [inline services/core-web/common/components/Step.tsx:8]: Yay TS!

File: services/core-web/common/components/SteppedForm.js (hunk 1 above)
Comment (matbusby-fw) [reply services/core-web/common/components/SteppedForm.js:13]: Weird!  Well we can convert it to TS eventually and use an interface.

**Review by matbusby-fw — APPROVED** [review]
🚀 🐫

**Pull request merged**